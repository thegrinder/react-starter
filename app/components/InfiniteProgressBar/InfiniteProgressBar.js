import styled, { keyframes } from 'styled-components';

const increase = keyframes`
  from { left: -15%; width: 5%; }
  to { left: 130%; width: 100%; }
`;

const decrease = keyframes`
  from { left: -80%; width: 80%; }
  to { left: 110%; width: 10%; }
`;

export const InfiniteProgressBar = styled.div`
  position: relative;
  height: 5px;
  overflow-x: hidden;
  background: #2196f3;
  &:before,
  &:after {
    content: '';
    position: absolute;
    background: #e3f2fd;
    height: 5px;
    width: 10%;
    left: -15%;
  }
  &:after {
    animation: ${increase} 2s infinite;
  }
  &:before {
    animation: ${decrease} 2s 0.5s infinite;
  }
`;
