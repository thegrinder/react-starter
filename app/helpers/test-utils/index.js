import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';
import { render } from 'react-testing-library';

import theme from '../../theme/components';

const propTypes = {
  children: PropTypes.node.isRequired,
};

const Providers = ({ children }) => (
  <IntlProvider locale="en">
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </IntlProvider>
);

Providers.propTypes = propTypes;

const customRender = (ui, options) => render(ui, { wrapper: Providers, ...options });

export * from 'react-testing-library';
export { customRender as render };
