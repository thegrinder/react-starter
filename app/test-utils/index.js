/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';
import * as RTL from '@testing-library/react';

import theme from '../theme';

const propTypes = {
  children: PropTypes.node.isRequired,
};

const Providers = ({ children }) => (
  <IntlProvider locale="en">
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </IntlProvider>
);

Providers.propTypes = propTypes;

export const render = (ui, options) =>
  RTL.render(ui, { wrapper: Providers, ...options });

export const { fireEvent, act } = RTL;

export const createStore = configureStore();
