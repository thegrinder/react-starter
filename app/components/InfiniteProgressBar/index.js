import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { getColor } from 'basic-styled-uikit';
import { rem } from '../../helpers/utils';

const increase = keyframes`
  from { left: -15%; width: 5%; }
  to { left: 130%; width: 100%; }
`;

const decrease = keyframes`
  from { left: -80%; width: 80%; }
  to { left: 110%; width: 10%; }
`;

const propTypes = {
  color: PropTypes.string,
  bgColor: PropTypes.string,
};

const defaultProps = {
  color: 'primary100',
  bgColor: 'primary500',
};

const InfiniteProgressBar = styled.div`
  position: relative;
  height: ${rem(5)};
  overflow-x: hidden;
  background: ${({ theme, bgColor }) => getColor(bgColor)({ theme })};
  &:before,
  &:after {
    content: '';
    position: absolute;
    background: ${({ theme, color }) => getColor(color)({ theme })};
    height: ${rem(5)};
    width: 10%;
    left: -15%;
  }
  &:after {
    animation: ${increase} 2s infinite;
  }
  &:before {
    animation: ${decrease} 2s 0.5s infinite;
  }
`;

InfiniteProgressBar.propTypes = propTypes;
InfiniteProgressBar.defaultProps = defaultProps;

export default InfiniteProgressBar;
