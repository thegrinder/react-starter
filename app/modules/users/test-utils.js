import { initialRequestState } from '../helpers';
import { initialUsersState } from './reducer';

export const createUser = (user = {}) => ({
  id: 'id',
  name: 'name',
  username: 'username',
  email: 'email@email.com',
  phone: '132456776',
  ...user,
});

export const createRequestState = (requestState = {}) => ({
  ...initialRequestState,
  ...requestState,
});

export const createUsersState = ({
  requestKey = '',
  requestState = {},
  data = {},
} = {}) => ({
  ...initialUsersState,
  data: {
    ...initialUsersState.data,
    ...data,
  },
  requests: {
    ...initialUsersState.requests,
    ...(requestKey && {
      [requestKey]: {
        ...initialUsersState.requests[requestKey],
        ...requestState,
      },
    }),
  },
});

export const createState = args => ({
  users: createUsersState(args),
});
