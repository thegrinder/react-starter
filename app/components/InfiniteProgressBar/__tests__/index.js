import React from 'react';

import { render } from 'test-utils';
import InfiniteProgressBar from '..';

const renderComponent = (props = {}) => render(<InfiniteProgressBar {...props} />);

describe('<InfiniteProgressBar />', () => {
  it('should render correctly with default props', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should render correctly with custom props', () => {
    const { container: { firstChild } } = renderComponent({ color: 'white', bgColor: 'black' });
    expect(firstChild).toBeDefined();
  });

  it('should render <div> tag', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild.tagName).toEqual('DIV');
  });
});
