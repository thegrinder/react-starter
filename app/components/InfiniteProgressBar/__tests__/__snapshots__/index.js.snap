// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfiniteProgressBar /> should render correctly with default props 1`] = `
<div
  class="sc-bdVaJa kxYSrp"
  color="primary100"
/>
`;
